src/main/java/com/sda/weather/entry/LocationService.java

```diff
@@ -1,4 +1,43 @@
 package com.sda.weather.entry;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+//warstwa logiki biznesowej
 public class LocationService {
+
+    // todo develop this class
+
+    private final LocationRepository locationRepository;
+    private final ObjectMapper objectMapper;
+
+    public LocationService(LocationRepository locationRepository, ObjectMapper objectMapper) {
+        this.locationRepository = locationRepository;
+        this.objectMapper = objectMapper;
+    }
+
+    Location createLocation(String city, String country, Double longitude, Double latitude) {
+        if (city == null || city.isBlank()) {
+            throw new IllegalArgumentException(" City cannot be null");
+        }
+        if (country == null || country.isBlank()) {
+            throw new IllegalArgumentException(" Country cannot be null");
+        }
+        if (longitude < -90D || longitude > 90D) {
```

Comment:
But longitude lines run north-south. They converge at the poles. And its X-coordinates are between -180 and +180 degrees.